### PR TITLE
Versioning Releases wiith Rever

### PR DESCRIPTION
**Pull request by GbotemiB**
This PR contains creating/managing version num on zarr with rever.

Steps:

after making changes in development. To update the version with rever, simply run the following commands in root directory.
$ rever setup
$ rever `<the version number>`

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

Fixes https://github.com/zarr-developers/zarr-python/issues/829

### CODE REVIEW

**Comment by joshmoore** [comment]
Thanks for looking into this, @GbotemiB. Can you update the description with the steps that someone would take to _use_ `rever`?

**Review by Saransh-cpp — COMMENTED** [review]
(I am not a maintainer. You can ignore my reviews if the maintainers approve of this. I generally like reading others' code and working on open-source research software 😄)

**Comment by joshmoore** [comment]
:+1: for @Saransh-cpp's review.

File: .gitignore [hunk 1]
```diff
@@ -1,76 +1,3 @@
-# Byte-compiled / optimized / DLL files
-__pycache__/
-*.py[cod]
```
[the remark this answers is not in the document]
Comment (GbotemiB) [reply .gitignore:3]: This was restoring the gitnore files. rever does not edit the .gitignore's content

**Comment by GbotemiB** [comment]
Since SCM manages the version on zarr. It wont be necessary to use rever to bump the versions. But if there is a need to use rever to bump versions, I have the steps outlined here.

**Comment by jakirkham** [comment]
Thanks for the work on this @GbotemiB! 🙏

Would it be possible to incorporate this into [our GitHub Actions release process]( https://github.com/zarr-developers/zarr-python/blob/e8f6fef0c737c167b16cd10accdcc3bbc6dea566/.github/workflows/releases.yml )?

Also do you think this would be possible to add to [Numcodecs]( https://github.com/zarr-developers/numcodecs )?

**Comment by GbotemiB** [comment]
> Thanks for the work on this @GbotemiB! pray
> 
> Would it be possible to incorporate this into [our GitHub Actions release process](https://github.com/zarr-developers/zarr-python/blob/e8f6fef0c737c167b16cd10accdcc3bbc6dea566/.github/workflows/releases.yml)?
> 
> Also do you think this would be possible to add to [Numcodecs](https://github.com/zarr-developers/numcodecs)?

[Re-ver](https://github.com/regro/rever) has support for Github releases. I will look into this.

I will look into bringing it to [Numcodecs](https://github.com/zarr-developers/numcodecs)

**Comment by jakirkham** [comment]
Definitely [we use rever in conda-forge]( https://github.com/conda-forge/conda-smithy/blob/e65bd1cfef08ca3da41c2b8f07d2be539f123fe7/README.md#releasing-conda-smithy ).

Ideally we would include this as a step in GitHub Actions (maybe [after this one]( https://github.com/zarr-developers/zarr-python/blob/fcd130628ba41e43dfb73d52aaa2318339dfd887/.github/workflows/releases.yml#L24-L26 )?). Maybe we would change the trigger (like using a bot command or issue).

There's also some discussion about options for changelog generation in issue ( https://github.com/zarr-developers/zarr-python/issues/829 ). From it a few things (including [rever's new workflow]( https://regro.github.io/rever-docs/news.html )) come up. Some of the others are:

* [towncrier]( https://towncrier.readthedocs.io/en/stable/quickstart.html )
* [sphinx-github-changelog]( https://sphinx-github-changelog.readthedocs.io/en/latest/ )
* [scriv]( https://scriv.readthedocs.io/en/latest/ )

**Comment by joshmoore** [comment]
> Would it be possible to incorporate this into [our GitHub Actions release process](https://github.com/zarr-developers/zarr-python/blob/e8f6fef0c737c167b16cd10accdcc3bbc6dea566/.github/workflows/releases.yml)?

@jakirkham: do you mean `rever` in general or would you replace `setuptools-scm`?

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshmoore) [reply .gitignore:3]: @GbotemiB: it still seems like this change is too large. Can _just_ `rever/` be added?

**Comment by jakirkham** [comment]
Sorry for missing this. What I mean is we can basically do a release from the GitHub UI today. That's incredibly valuable as that keeps things easy for others wanting to release. Think any changes we make to the release log process, should preserve that functionality. Otherwise we are merely trading one pain point for another (simpler changelog management for harder release process)

**Comment by jhamman** [comment]
This has unfortunately gone stale. Will it be revived or should we close this?

**Pull request closed without merging**